src/sidebar/store/modules/real-time-updates.ts

```diff
@@ -221,6 +223,48 @@ function hasPendingUpdatesOrDeletions(state: State): boolean {
   return pendingUpdateCount(state) > 0;
 }
 
+type RootState = {
+  realTimeUpdates: State;
+  session: SessionState;
+  annotations: AnnotationsState;
+};
+
+/**
+ * Return a total count of pending updates containing mentions for current user.
+ * This includes any new annotation where current user was mentioned, or any
+ * existing annotation which was edited to add a new mention to current user.
+ */
+const pendingMentionCount = createSelector(
+  (rootState: RootState) => rootState.realTimeUpdates.pendingUpdates,
+  (rootState: RootState) => sessionModule.selectors.profile(rootState.session),
+  (rootState: RootState) =>
+    annotationsModule.selectors.allAnnotations(rootState.annotations),
+  (pendingUpdates, { userid: currentUserId }, allAnnotations) => {
+    const pendingAnnotations = Object.values(pendingUpdates);
+    const mentionsIncludeCurrentUser = (mentions?: Mention[]) =>
+      !!mentions?.some(mention => mention.userid === currentUserId);
+
+    return pendingAnnotations.filter(({ mentions, id }) => {
+      const mentionToCurrentUserIsNowIncluded =
```

Comment:
```suggestion
      const mentionOfCurrentUserIsNowIncluded =
```

I would have gone for a shorter name like `mentionsCurrentUser`.